tools/system_libs.py

```diff
@@ -549,30 +549,59 @@ def get_default_variation(cls, **kwargs):
 
 class NoExceptLibrary(Library):
   def __init__(self, **kwargs):
-    self.is_noexcept = kwargs.pop('is_noexcept')
+    self.eh_mode = kwargs.pop('eh_mode')
```

Comment:
I think the code might be simpler without adding `self.eh_mode`. Can we just always check the two `shared.Settings.*` flags? Maybe we can add helper functions `using_emscripten_exceptions()` and `using_wasm_exceptions()` to simplify that.

Reply:
It is mostly just for readability. And without that this kind of code (which is a `classmethod` in `class NoExceptLibrary`) will look uglier:
```
  @classmethod
  def variations(cls, **kwargs):
    combos = super(NoExceptLibrary, cls).variations()
    return ([dict(eh_mode='noexcept', **combo) for combo in combos] +
            [dict(eh_mode='emscripten', **combo) for combo in combos] +
            [dict(eh_mode='wasm', **combo) for combo in combos])
```

Reply:
I see, yeah, using `eh_mode` that way is nice.

How about avoiding the string literals that are scattered around? That feels a little risky to me for typos, and we don't seem to use that pattern in general (except for emmalloc, but that arrives from `shared.Settings.MALLOC` so it sort of makes sense there). Can do something like
```python
class exceptions:
  # explain what none is
  none = 'none'
  emscripten = 'emscripten'
  wasm = 'wasm'
```
and then use `eh_mode=exceptions.wasm` etc. There might be a nicer python pattern though.

Reply:
Members don't need to be strings after all? Changed to:
```
class exceptions(object):                                                        
  none = 0 # Don't use exception support                                         
  emscripten = 1                                                                 
  wasm = 2  
```